From 92e8e410fbd6c4e36cdb8635a0139db879ba790b Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:22 -0400
Subject: [tools] perf/stat: Add units to nanosec-based counters

Message-id: <1408462094-14194-154-git-send-email-jolsa@redhat.com>
Patchwork-id: 88115
O-Subject: [PATCH RHEL7.1 BZ1131394 153/325] perf stat: Add units to nanosec-based counters
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4bbe5a61f29b13437a6a16467328d3bae8fce9e7
Author: David Ahern <dsahern@gmail.com>
Date: Sat Sep 28 14:28:00 2013 -0600

description
===========
Ingo pointed out that the task-clock counter should have the units
explicitly stated since it is not a counter.

Before:

perf stat -a -- sleep 1

 Performance counter stats for 'sleep 1':

      16186.874834 task-clock          #   16.154 CPUs utilized
...

After:

perf stat -a -- sleep 1

 Performance counter stats for 'system wide':

      16146.402138 task-clock (msec)   #   16.125 CPUs utilized
...
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-stat.c b/tools/perf/builtin-stat.c
index 2178e66..1a9c95d 100644
--- a/tools/perf/builtin-stat.c
+++ b/tools/perf/builtin-stat.c
@@ -706,10 +706,13 @@ static void nsec_printout(int cpu, int nr, struct perf_evsel *evsel, double avg)
 {
  double msecs = avg / 1e6;
  const char *fmt = csv_output ? "%.6f%s%s" : "%18.6f%s%-25s";
+ char name[25];
 
  aggr_printout(evsel, cpu, nr);
 
- fprintf(output, fmt, msecs, csv_sep, perf_evsel__name(evsel));
+ scnprintf(name, sizeof(name), "%s%s",
+    perf_evsel__name(evsel), csv_output ? "" : " (msec)");
+ fprintf(output, fmt, msecs, csv_sep, name);
 
  if (evsel->cgrp)
   fprintf(output, "%s%s", csv_sep, evsel->cgrp->name);
-- 
1.7.1